projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -977,4 +983,14 @@ export class IgxOverlayService implements OnDestroy {
             info.closeAnimationPlayer.finish();
         }
     }
+
+    private addComponentSize(info: OverlayInfo) {
+        if (info?.elementRef?.nativeElement && info.elementRef.nativeElement instanceof Element) {
+            const styles = window.getComputedStyle(info.elementRef.nativeElement);
```

Comment:
Perhaps unlikely scenario for attach to run on some component's init in SSR, but not entirely impossible. Even for consistency, we should make it a habit to avoid directly accessing `window` properties.
`globalThis.window?.getComputedStyle` is one way used widely (there's even `getComponentSize` in utils that does the next line like that, tho only used in.. tests for some reason. Alt you can prob also use the injected doc token (`this._document.defaultView`) though need to see if a check is required for the method existence in domino